File: Java Self/x1.java
class x2 {
    int x = 10;

    void p() {
        int x = 20;
        System.out.println(x);
        System.out.println(this.x);
    }
}

class x1 {
    public static void main(String[] args) {
        x2 obj = new x2();
        obj.p();
    }
}

File: C++ Self/x1.cpp
#include <iostream>
using namespace std;
class x1
{
public:
    template <typename T>
    T add(T a, T b)
    {
        return a + b;
    }
};
int main()
{
    x1 obj;
    cout << obj.add(10, 20) << endl;
    cout << obj.add(10.5, 20.5) << endl;
    return 0;
}
